### PR TITLE
Convert shared annotator/ types to TypeScript

### PR DESCRIPTION
**Pull request by robertknight**
This is a purely syntactic change. The documentation and types are the same as before. One thing to note about using TS syntax is that a module only exports types which are explicitly marked for export, whereas in JSDoc any types defined in the module or imported from elsewhere are exported.

### CODE REVIEW

**Comment by lyzadanger** [comment]
> This is a purely syntactic change. The documentation and types are the same as before. 

Well, except for my local-branch changes to this module :P — I'll see how nasty the merge is...

**Comment by robertknight** [comment]
There is a lint issue complaining about referencing types before their declaration. This is just a stylistic preference as TypeScript allows types to be declared after references in a module, so I'm just going to look into disabling this warning.

**Comment by robertknight** [comment]
> Well, except for my local-branch changes to this module :P — I'll see how nasty the merge is...

Ah, sorry about that. It will probably be easiest to manually re-implement the changes in TS once this lands.

**Comment by lyzadanger** [comment]
> Ah, sorry about that. It will probably be easiest to manually re-implement the changes in TS once this lands.

Oh, no worries! Just goes to show we're busy bees right now! Let's work on moving this forward so we can get it landed and I can do what I need to do on my branch!

File: src/annotator/integrations/vitalsource.js [hunk 1]
```diff
@@ -12,8 +12,8 @@ import { injectClient } from '../hypothesis-injector';
  * @typedef {import('../../types/annotator').Anchor} Anchor
  * @typedef {import('../../types/annotator').Annotator} Annotator
  * @typedef {import('../../types/annotator').Integration} Integration
- * @typedef {import('../../types/annotator').Selector} Selector
  * @typedef {import('../../types/annotator').SidebarLayout} SidebarLayout
+ * @typedef {import('../../types/api').Selector} Selector
```
Comment (lyzadanger) [inline src/annotator/integrations/vitalsource.js:16]: ...just...ordering?

**Review by lyzadanger — APPROVED** [review]
All righty, let's get this merged and I'll unroll my stuff into it locally!

File: src/annotator/integrations/vitalsource.js (hunk 1 above)
Comment (robertknight) [reply src/annotator/integrations/vitalsource.js:16]: The `Selector` type is declared in `types/api.js` but used to be implicitly re-exported by `types/annotator.js`. That implicit re-exporting no longer happens.

**Pull request merged**